src/System.Net.WebHeaderCollection/src/System/Net/WebHeaderCollection.cs

```diff
@@ -96,45 +139,251 @@ private bool AllowHttpResponseHeader
             }
         }
 
-        public string this[string name]
+        public override void Set(string name, string value)
         {
-            get
+            if (string.IsNullOrEmpty(name))
             {
-                string entry = null;
-                _entriesDictionary?.TryGetValue(name, out entry);
-                return entry;
+                throw new ArgumentNullException(nameof(name));
             }
-            set
+
+            name = HttpValidationHelpers.CheckBadHeaderNameChars(name);
+            ThrowOnRestrictedHeader(name);
+            value = HttpValidationHelpers.CheckBadHeaderValueChars(value);
+            GlobalLog.Print("WebHeaderCollection::Set() calling InnerCollection.Set() key:[" + name + "], value:[" + value + "]");
```

Comment:
Please see examples of the GlobalLog usage after the code got ported: https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/Common/src/System/Net/InternalException.cs#L13

This (and all other calls) needs to be replaced by:
```C#
            if (GlobalLog.IsEnabled)
            {
                 GlobalLog.Print("...
            }
```